include/sys/arch_interface.h

```diff
@@ -343,6 +343,32 @@ int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
  * @see ISR_DIRECT_DECLARE()
  */
 
+#ifndef CONFIG_PCIE_CONTROLLER
+/**
+ * Arch-specific hook for allocating IRQs
```

Comment:
I suggest to put @brief here. 
This is interesting but I am not sure if this interface can be generic enough to fit other platform. But it's a good try I think.

Reply:
As long as there are no other arch using it atm but x86 I can't tell what other parameters would be required. It will be fine to change the signature when relevant (i.e. when another ARCH will require it).